### PR TITLE
fix(eslint): check whole function scopes for used variables

### PR DESCRIPTION
**Pull request by LukBrezina**
I've noticed your rule is covering just parent qHook/qComponent params,  from the examples Miško provided I think it should consider whole scope. 

AST explorer with scoped variables: 
https://astexplorer.net/#/gist/55f073408c4a5d50f051a2e40830abff/adaeba3d56546020165fd37e74f38e41e025941b

### CODE REVIEW

**Comment by otodockal** [comment]
@luki215 yeah, definitely the whole scope, you're right! thank you.

**Review by otodockal — APPROVED** [review]

**Comment by otodockal** [comment]
@luki215 just one thing, could you pls update the PR and only rename this part (and related code)?

from:
```js
qHookStore.set(getNodeId(node), {
  params: flattenFnNodeParams(fnNode),
  parentNode: walkToFirstQHook(node.parent),
});
```  

to:
```js
qHookStore.set(getNodeId(node), {
  variables: flattenScopeVariables(fnNode),
  parentNode: walkToFirstQHook(node.parent),
});
```                  

Or something like that. Sorry for this request, but it's no longer about params.

**Review by otodockal — CHANGES_REQUESTED** [review]
one update pls according to my prev comment, thank you.

**Comment by LukBrezina** [comment]
> one update pls according to my prev comment, thank you.

Yeah, I've missed that, thank you! Updated.

File: src/eslint/index.js [hunk 1]
```diff
@@ -102,7 +93,7 @@ module.exports = {
                 if (fnNode) {
                   // flatten qHook references in order to easily access params & parentNode in reporting
                   qHookStore.set(getNodeId(node), {
-                    params: flattenFnNodeParams(fnNode),
+                    params: flattenScopeVariables(fnNode),
```
Comment (otodockal) [inline src/eslint/index.js]: @luki215 and last thing, params -> **variables**. thanks. sorry :-D

Comment (LukBrezina) [reply src/eslint/index.js]: :D done 👍

**Pull request merged**